pkg/network/server_connection.go

```diff
@@ -196,17 +209,8 @@ func (t *serverTcpConnection) startAcceptPolling(closedServerChannel chan struct
 		}
 	}()
 	go func() {
-		<-t.ctx.Done()
-		t.logger.Infof("Closing control server")
-		err := t.listener.Close()
-		t.logger.Infof("Listener closed")
-		if err != nil {
-			t.logger.Warnf("Error while closing the listener: %s", err)
-		}
-		err = t.close()
-		if err != nil {
-			t.logger.Warnf("Error while closing the tcp connection: %s", err)
-		}
+		<-stoppedConsumingCtx.Done()
+		t.close()
 		close(closedServerChannel)
```

Comment:
You can ditch the `stoppedConsumingCtx` and move these pieces into the `case <-t.ctx.Done():` select case above I believe. Spares you are goroutine and a channel.